static/app/components/searchQueryBuilder/input.tsx

```diff
@@ -215,7 +215,7 @@ function SearchQueryBuilderInputInternal({
   tabIndex,
   state,
 }: SearchQueryBuilderInputInternalProps) {
-  const trimmedTokenValue = token.value.trim();
+  const trimmedTokenValue = token.text.trim();
```

Comment:
Only difference is that text includes quotes, which I noticed was necessary if you want to type `"foo and bar"` without `and` being tokenized